graphql/schemabuilder/helpers.go

```diff
@@ -390,3 +405,111 @@ func castSlice(slice interface{}) []interface{} {
 
 	return ret
 }
+
+func (sb *schemaBuilder) buildPaginatedArgParser(originalArgType reflect.Type) (*argParser, graphql.Type, error) {
+
+	interfaceParser, interfaceType, err := sb.makeStructParser(originalArgType)
```

Comment:
This function could you some comments interspersed between its different steps. maybe worth adding one for the function itself as well.